### DISCUSSION TITLE
Disable count to improve performance on large dataset

### DISCUSSION THREAD

**Discussion by asafhbinov** [alirezanet/Gridify#172] [Q&A] [answered]
Hi, I have a complex data structure on a large data set, and I would like to disable the count(*) query for optimization.
Is there a way to do this or provide results from another data source (for example a pre-calculated field)
I.e separate the count(*) from the actual paged data.

**Comment by alirezanet** [comment, marked as answer]
Hi @asafhbinov,
if you check the `GridifyQueryable` method, it essentially applies filtering, retrieves the count, and then ordering and paging sequentially. 

```csharp
   public static QueryablePaging<T> GridifyQueryable<T>(this IQueryable<T> query, IGridifyQuery? gridifyQuery, IGridifyMapper<T>? mapper = null)
   {
      query = query.ApplyFiltering(gridifyQuery, mapper);
      var count = query.Count();
      query = query.ApplyOrdering(gridifyQuery, mapper);
      query = query.ApplyPaging(gridifyQuery);
      return new QueryablePaging<T>(count, query);
   }
```

You have complete control over each operation within Gridify. You can create your own extension to bypass the counting process, or you can opt to use the `ApplyFilteringOrderingPaging` method directly.

```csharp
   public static IQueryable<T> ApplyFilteringOrderingPaging<T>(this IQueryable<T> query, IGridifyQuery? gridifyQuery,
      IGridifyMapper<T>? mapper = null)
   {
      if (gridifyQuery == null) return query;

      query = query.ApplyFiltering(gridifyQuery, mapper);
      query = query.ApplyOrdering(gridifyQuery, mapper);
      query = query.ApplyPaging(gridifyQuery);
      return query;
   }
```
